docs/user-guide/connecting-applications.md

```diff
@@ -393,6 +393,165 @@ LoadBalancer Ingress:   a320587ffd19711e5a37606cf4a74574-1142138393.us-east-1.el
 ...
 ```
 
+## Loadbalancing multiple Services
+
+Till now we had restricted ourselves to a single Service, but Kubernetes has a resource called [Ingress](/docs/user-guide/ingress) that spans Services and loadbalances requests at [layer 7](https://en.wikipedia.org/wiki/OSI_model#Layer_7:_Application_Layer). This is useful for reasons like:
+
+* Cost saving, since the Ingress only needs one public IP, instead of 1 per Service of `Type=LoadBalancer`.
+* Centralized security configuration, instead of loading TLS certs as volumes in each Service backend.
+* Smarter routing, since the Ingress has visibility into layer 7 of the OSI model.
+
+For demonstration purposes we will deploy a second Service that simply echoes the HTTP Headers it receives:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoheaders
+  labels:
+    app: echoheaders
+spec:
+  type: NodePort
+  ports:
+  - port: 80
+    targetPort: 8080
+    protocol: TCP
+    name: http
+  selector:
+    app: echoheaders
+---
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: echoheaders
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: echoheaders
+    spec:
+      containers:
+      - name: echoheaders
+        image: gcr.io/google_containers/echoserver:1.2
+        ports:
+        - containerPort: 8080
+```
+
+To make use of the Ingress resource, you need to first deploy an Ingress controller. On GCE/GKE there should already be an l7-controller deployed into the `kube-system` namespace:
+
+```shell
+$ kubectl get pods --namespace=kube-system -l name=glbc
+NAME                            READY     STATUS    RESTARTS   AGE
+l7-lb-controller-v0.6.0-chnan   2/2       Running   0          1d
+```
+
+Since Ingress is still a beta resource, make sure you review the [beta limitations](https://github.com/kubernetes/contrib/tree/master/ingress/controllers/gce/BETA_LIMITATIONS.md) of the controller. In particular, you need to create a single firewall-rule on your cloudprovider, to allow health checks. On GKE this would be:
+
+```shell
+$ export TAG=$(basename `gcloud container clusters describe ${CLUSTER_NAME} --zone ${ZONE} | grep gke | awk '{print $2}'` | sed -e s/group/node/)
+$ export NODE_PORT=$(kubectl get -o jsonpath="{.spec.ports[0].nodePort}" services echoheaders)
+$ gcloud compute firewall-rules create allow-130-211-0-0-22 \
+  --source-ranges 130.211.0.0/22 \
+  --target-tags $TAG \
+  --allow tcp:$NODE_PORT
+```
+
+In environments other than GCE/GKE, you need to [deploy a controller](https://github.com/kubernetes/contrib/tree/master/ingress/controllers).
+Once your controller is Running, you can create an Ingress:
+
+```yaml
```

Comment:
It would be so great if you could add this as an external yaml file and use the {% include code.html ...%} method documented in README.md at the root of this repo to include it. That way, users can download this file directly and/or copy its contents into their clipboard.